### PR TITLE
[core] yaml anchor-alias support for restapi

### LINKED ISSUE #16315: Investiguate YAML anchor support
### Describe the issue

The following flow YAML contains an anchor; it is correctly validated, but when trying to execute the flow, the inputs are not correctly defined.

```yaml
id: test_anchor_alias
namespace: company.team

inputs:
  - id: foo
    type: STRING
    defaults: &myvalue blah
  - id: blah
    type: STRING
    defaults: *myvalue

tasks:
  - id: hello
    type: io.kestra.plugin.core.log.Log
    message: "{{ inputs.foo }} {{ inputs.blah }}"
```

The following snippet shows that the anchor didn't work, and when executing the flow, the pebble expression wrongly evaluates to myvalue for `{{ inputs.blah }}`.

Image

### Environment

- Kestra Version: develop

### PR DESCRIPTION
**Pull request by commiter-atharvakamat**
### ✨ Description

In reference to issue id #16315
Edit: closes #16315 

What does this PR change?  

### 🔗 Related Issue

YAML anchor support for Flow controller, yamlparser and plugindefaultservice components.

### 🛠️ Backend Checklist

YamlParser Utility: Modified the private read() and added overloaded helper to seamlessly handle both explicit classes (Class<T>) and generic collection frameworks (TypeReference<T>). Updated internal static mappers to execute convertValue safely against pre-parsed object streams.

PluginDefaultService: Updated the parseFlowWithAllDefaults() transformations immediately before persistence to leverage the new anchor-safe utility.

- [x] Code compiles successfully and passes all checks
- [x] All unit and integration tests pass

### 📝 Additional Notes

Add any extra context or details reviewers should be aware of.

### CODE REVIEW

**Comment by commiter-atharvakamat** [comment]
failing checks
here are the discussion links to the previously encountered issues.
https://stackoverflow.com/questions/40074700/jackson-yaml-support-for-anchors-and-references
https://groups.google.com/g/jackson-user/c/b3ucK2emRbE?pli=1

get and delete request errors.
errors are when mapFlow as null.
others when the revision mismatch.

tests failing and this is one of the errors.

    io.kestra.repository.h2.H2FlowRepositoryTest ✘ findByExecution()

    java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0
        at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:100)
        at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:106)
        at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:302)
        at java.base/java.util.Objects.checkIndex(Objects.java:365)
        at java.base/java.util.ArrayList.get(ArrayList.java:428)
        at org.yaml.snakeyaml.scanner.ScannerImpl.peekToken(ScannerImpl.java:278)
        at org.yaml.snakeyaml.parser.ParserImpl$ParseDocumentEnd.produce(ParserImpl.java:279)
        at org.yaml.snakeyaml.parser.ParserImpl.peekEvent(ParserImpl.java:161)
        at org.yaml.snakeyaml.comments.CommentEventsCollector$1.peek(CommentEventsCollector.java:57)
        at org.yaml.snakeyaml.comments.CommentEventsCollector$1.peek(CommentEventsCollector.java:43)
        at org.yaml.snakeyaml.comments.CommentEventsCollector.collectEvents(CommentEventsCollector.java:136)
        at org.yaml.snakeyaml.comments.CommentEventsCollector.collectEvents(CommentEventsCollector.java:116)
        at org.yaml.snakeyaml.composer.Composer.composeScalarNode(Composer.java:249)
        at org.yaml.snakeyaml.composer.Composer.composeNode(Composer.java:214)
        at org.yaml.snakeyaml.composer.Composer.getNode(Composer.java:141)
        at org.yaml.snakeyaml.composer.Composer.getSingleNode(Composer.java:167)
        at org.yaml.snakeyaml.constructor.BaseConstructor.getSingleData(BaseConstructor.java:179)
        at org.yaml.snakeyaml.Yaml.loadFromReader(Yaml.java:508)
        at org.yaml.snakeyaml.Yaml.load(Yaml.java:437)
        at io.kestra.core.serializers.YamlParser.parseAnchorAndAlias(YamlParser.java:125)
        at io.kestra.core.serializers.YamlParser.read(YamlParser.java:101)
        at io.kestra.core.serializers.YamlParser.parse(YamlParser.java:43)
        at io.kestra.core.models.flows.GenericFlow.fromYaml(GenericFlow.java:74)
        at io.kestra.core.models.flows.GenericFlow.of(GenericFlow.java:63)
        at io.kestra.core.repositories.AbstractFlowRepositoryTest.findByExecution(AbstractFlowRepositoryTest.java:1042)
        at java.base/java.lang.reflect.Method.invoke(Method.java:565)
        at io.micronaut.test.extensions.junit5.MicronautJunit5Extension$2.proceed(MicronautJunit5Extension.java:154)
        at io.micronaut.test.extensions.AbstractMicronautExtension.interceptEach(AbstractMicronautExtension.java:171)
        at io.micronaut.test.extensions.AbstractMicronautExtension.interceptTest(AbstractMicronautExtension.java:128)
        at io.micronaut.test.extensions.junit5.MicronautJunit5Extension.interceptTestMethod(MicronautJunit5Extension.java:141)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:511)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1450)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2019)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:187)

**Comment by commiter-atharvakamat** [comment]
Hi @loicmathieu 

I have fixed the YAML anchor and alias string input parsing issues in the backend, ensuring full compatibility with existing default configurations.

The new commit adds the necessary input validations before parsing and resolves the edge cases where earlier tests were failing. Additionally, the new SnakeYAML instance is now isolated within the YamlParser.parseAnchorAndAlias() method.

All related checks are now passing cleanly, including 
:jdbc-h2:test and 
:core:unitTest.

Please review my changes.

Edit: I explicitly added logic to return null for null inputs and throw an exception for empty inputs during the load phase and then converting it to the object class in this new commit. This is because it restores the exact behavior of the original ObjectMapper.readValue(), which the test suite requires to pass.

**Comment by loicmathieu** [comment]
@atharva-kamat can you rebase your changes onto master and rebuild to be sure no source format are changed?

**Comment by commiter-atharvakamat** [comment]
@loicmathieu Done! I have rebased against master/develop and successfully ran spotlessApply locally to ensure the source formatting aligns with Kestra's standards. Passing tests locally.

**Comment by commiter-atharvakamat** [comment]
Hello @loicmathieu Could you please review my work.

**Comment by commiter-atharvakamat** [comment]
Hi @loicmathieu 

I have successfully added YAML anchor and alias support to the flow-reading logic, ensuring full adherence to the new processing and parsing frameworks. The implementation now correctly handles these references up front before running through the strict mapping and validation layers.

I have attached a video demonstrating the updated implementation in action and confirming the successful resolution of all the provided edge cases.

Please let me know if you have any feedback or questions!


https://github.com/user-attachments/assets/e2b74db6-cdfd-42e7-8284-a58f27285aa1

File: core/src/main/java/io/kestra/core/services/FlowParsingService.java [hunk 1]
```diff
@@ -144,10 +144,14 @@ public static FlowWithSource toFlowWithSource(final FlowInterface flow) {
 
         // The block below should only be reached during testing for failure scenarios
         try {
-            Flow parsed = YAML_MAPPER_NON_DEFAULT.readValue(flow.getSource(), Flow.class);
+            // Flow parsed = YAML_MAPPER_NON_DEFAULT.readValue(flow.getSource(), Flow.class);
```
Comment (loicmathieu) [inline core/src/main/java/io/kestra/core/services/FlowParsingService.java]: Can you remove this commented line? The same from line 206

File: core/src/main/java/io/kestra/core/services/FlowParsingService.java [hunk 2]
```diff
@@ -144,10 +144,14 @@ public static FlowWithSource toFlowWithSource(final FlowInterface flow) {
...
+            Flow parsed = YamlParser.parse(flow.getSource(), Flow.class);
             return FlowWithSource.of(parsed, flow.getSource());
-        } catch (JsonProcessingException e) {
-            throw new KestraRuntimeException("Failed to read flow from source", e);
+        } catch (Exception e) {
+            if (e instanceof JsonProcessingException je) {
+                throw new KestraRuntimeException("Failed to read flow from source", e);
+            }
+            throw e;
```
Comment (loicmathieu) [inline core/src/main/java/io/kestra/core/services/FlowParsingService.java]: Can you revert? catching exception is not a good idea.

File: core/src/main/java/io/kestra/core/serializers/YamlParser.java [hunk 3]
```diff
@@ -83,10 +89,38 @@ public static <T> T parse(File file, Class<T> cls) throws ConstraintViolationExc
 
     private static <T> T read(String input, Class<T> objectClass, String resource) {
         try {
-            return STRICT_MAPPER.readValue(input, objectClass);
-        } catch (JsonProcessingException e) {
-            throw toConstraintViolationException(input, resource, e);
+            Object parsedYaml = parseAnchorAndAlias(input);
+            // return STRICT_MAPPER.readValue(input, objectClass);
+            return STRICT_MAPPER.convertValue(parsedYaml, objectClass);
```
Comment (loicmathieu) [inline core/src/main/java/io/kestra/core/serializers/YamlParser.java:93]: This would resolves in a two-pass parsing. Can't you only parse once?

**Review by loicmathieu — CHANGES_REQUESTED** [review]
It also clearly miss a test

**Comment by commiter-atharvakamat** [comment]
I will make the required changes and submit it. Thanks!

File: core/src/main/java/io/kestra/core/serializers/YamlParser.java (hunk 3 above)
Comment (commiter-atharvakamat) [reply core/src/main/java/io/kestra/core/serializers/YamlParser.java:93]: I looked into this one. The thing is that we have another snakeyaml loadAs() method that requires string, class. But the concern is that it cannot fix the typereferences that are called later during execution.
They are jackson specific.
So the workaround i found is that we have the snakeyaml which loads the string directly as the object. we immediately convert it to required class or typereference. we use strict_mapper so the configurations and information remain intact i believe.
So, I thought this as a single step and then the required value conversions still take place in another components.

The traditional STRICT_Mapper.readValue() completely misses the anchor-alias.
I also searched and tweaked for whether we could add any configurations to the yaml factory on creation but it does not have those properties and methods.
I read on some thread that dependency upgradation might be a solution but I am not sure about it whether it works.

So, I fallback to the given solution.

File: core/src/main/java/io/kestra/core/services/FlowParsingService.java (hunk 1 above)
Comment (commiter-atharvakamat) [reply core/src/main/java/io/kestra/core/services/FlowParsingService.java]: Yes Sure!

**Comment by commiter-atharvakamat** [comment]
passing checks
./gradlew :jdbc-h2:test
./gradlew :core:unitTest
./gradlew :webserver:unitTest

however, i think that certain checks are failing due to changed edge cases and exceptions resolution by component.
./gradlew :webserver:test
